src/D2L.CodeStyle.Analyzers/Immutability/MutabilityInspector.cs

```diff
@@ -283,10 +284,46 @@ HashSet<ITypeSymbol> typeStack
 				}
 			}
 
+			// Inspect the type parameter as part of its host symbol

```

Comment:
It feels like these 2 blocks are methods on their own.  They are asking whether the generic type parameter is immutable.  

var hostType = hostSymbol as INamedTypeSymbol;
if( hostType != default ) {
  
     if( CheckHostTypeArguments => NotImmutable ) {
        return NotImmutable
     }

    if( CheckHostTypeInterfaceArguments => NotImmutable ) {
        return NotImmutable
     }
}